### PR TITLE
Performance: Optimized the recent files widget

### PR DESCRIPTION
**Pull request by jiejasonliu**
**Resolved / Related Issues**
- in general, this PR adds performance improvements through figuring out the type of change that occurs
  - more than often, changes only occur to one file (add/updated/removed); if it deviates, we can do a simple O(N) refresh
- also fixes a refresh bug from #10760
  - it seems that the prior solution falls short when a recent item is updated or when multiple changes occur, sometimes the OS batches changes to the recent items

**Validation**
How did you test these changes?
- [X] Built and ran the app
- [X] Tested the changes for accessibility

### CODE REVIEW

**Comment by jiejasonliu** [comment]
pinging @gave92 since you've been working around this area

File: src/Files.App/Filesystem/RecentItems.cs [hunk 1]
```diff
@@ -205,6 +206,49 @@ public Task<bool> UnpinFromRecentFiles(RecentItem item)
 			}));
 		}
 
+		private NotifyCollectionChangedEventArgs GetChangedActionEventArgs(IReadOnlyList<RecentItem> oldItems, IList<RecentItem> newItems)
+		{
+			var intersection = oldItems.Intersect(newItems);
```
Comment (gave92) [inline src/Files.App/Filesystem/RecentItems.cs:211]: I'm not sure this works that way you intend it to. Intersect returns the items which are in common between two lists, not the ones which are different.
Also RecentItem did not propertly implement IEquatable so `intersection` was always empty. I've fixed this in 
595ac69.

**Review by gave92 — CHANGES_REQUESTED** [review]

File: src/Files.App/Filesystem/RecentItems.cs (hunk 1 above)
Comment (gave92) [reply src/Files.App/Filesystem/RecentItems.cs]: Edit: I've fixed GetChangedActionEventArgs() in c6d3cb5 the way I think it's right. Could you take a look?

Comment (jiejasonliu) [reply src/Files.App/Filesystem/RecentItems.cs]: totally right, i meant set difference not intersection 😅 

your changes look right to me, thanks for taking a lot at this!

**Review by gave92 — APPROVED** [review]

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**